spec/counter_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::Counter do
+  describe '.initialize' do
+    it 'creates a counter' do
+      expect(Kitchen::Counter.new).to be_truthy
+    end
+  end
+  describe '.increment' do
+    counter = Kitchen::Counter.new
+    it 'increments by 1 given no args' do
+      val = counter.instance_variable_get(:@value)
+      counter.increment
+      expect(counter.instance_variable_get(:@value)).to eq(val + 1)
+    end
+    it 'increments by 2' do
+      val = counter.instance_variable_get(:@value)
+      counter.increment(by: 2)
+      expect(counter.instance_variable_get(:@value)).to eq(val + 2)
```

Comment:
This logic is totally right but there's a more compact way to write it

```ruby
expect { counter.increment(by: 2) }.to change { counter.get }.from(0).to(2)
```

See https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher

Reply:
Note that this passes blocks around (see the curly braces) instead of passing arguments using parentheses.  The code in the blocks is passed into `expect` so it can run the code when it is ready.  Internally it runs `counter.get` first to get the initial value, then it runs `counter.increment(by: 2)`, then it runs `counter.get` again